src/economy.cpp

```diff
@@ -1034,38 +1034,55 @@ static SmallIndustryList _cargo_delivery_destinations;
  */
 static uint DeliverGoodsToIndustry(const Station *st, CargoID cargo_type, uint num_pieces, IndustryID source)
 {
-	/* Find the nearest industrytile to the station sign inside the catchment area, whose industry accepts the cargo.
+	/* Find a random industry near the station sign inside the catchment area, whose industry accepts the cargo.
 	 * This fails in three cases:
 	 *  1) The station accepts the cargo because there are enough houses around it accepting the cargo.
 	 *  2) The industries in the catchment area temporarily reject the cargo, and the daily station loop has not yet updated station acceptance.
 	 *  3) The results of callbacks CBID_INDUSTRY_REFUSE_CARGO and CBID_INDTILE_CARGO_ACCEPTANCE are inconsistent. (documented behaviour)
 	 */
 
 	uint accepted = 0;
+	if (!st->industries_near.Length()) return accepted;
```

Comment:
Small personal nitpick: Personally I would prefer `if`-checks that test numerical values (instead of bools) to use proper comparisons (`== 0` in this case).
I do realise that the way it is written is valid C++ and what @PeterN suggested, so I'm certainly not going to veto the PR if it stays this way.

Reply:
Apologies, I suggested it but based on the previous code, which was a massive indentation block for no good reason, didn't check the return types.

I definitely agree with you!